src/main/java/io/sniffy/Sniffer.java

```diff
@@ -88,19 +89,43 @@ private static synchronized void notifyListeners(SocketMetaData socketMetaData,
     }
 
     public static void logSocket(String stackTrace, int connectionId, InetSocketAddress address, long elapsedTime, int bytesDown, int bytesUp) {
-        // increment counters
-        SocketMetaData socketMetaData = new SocketMetaData(address, connectionId, stackTrace, Thread.currentThread().getId());
+        
+        // do not track JDBC socket operations
+        SocketStats socketStats = sqlSocketStats.get();
+        if (null != socketStats) {
+            socketStats.accumulate(elapsedTime, bytesDown, bytesUp);
+        } else {
+            // increment counters
+            SocketMetaData socketMetaData = new SocketMetaData(address, connectionId, stackTrace, Thread.currentThread().getId());
+
+            // notify listeners
+            notifyListeners(socketMetaData, elapsedTime, bytesDown, bytesUp);
+        }
+    }
 
-        // notify listeners
-        notifyListeners(socketMetaData, elapsedTime, bytesDown, bytesUp);
+    private static ThreadLocal<SocketStats> sqlSocketStats = new ThreadLocal<SocketStats>();
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/sniffy/sniffy/file/2530680662/issues/source?bid=3005960&fileBranchId=3283367#l106)